chain/x/zoracle/exec_env.go

```diff
@@ -65,19 +86,47 @@ func (env *ExecutionEnvironment) GetValidatorAddress(validatorIndex int64) ([]by
 }
 
 func (env *ExecutionEnvironment) GetMaximumResultSize() int64 {
-	return env.keeper.MaxResultSize(env.ctx)
+	return env.maxResultSize
 }
 
 func (env *ExecutionEnvironment) GetMaximumCalldataOfDataSourceSize() int64 {
-	return env.keeper.MaxCalldataSize(env.ctx)
+	return env.maxCalldataSize
 }
 
 func (env *ExecutionEnvironment) RequestExternalData(
 	dataSourceID int64,
 	externalDataID int64,
 	calldata []byte,
-) error {
-	return env.keeper.AddNewRawDataRequest(env.ctx, env.requestID, types.ExternalID(externalDataID), types.DataSourceID(dataSourceID), calldata)
+) {
+	env.rawDataRequests = append(env.rawDataRequests, types.NewRawDataRequestWithExternalID(
```

Comment:
Let's make sure there is an upper bound on the slice length. Load max raw data request count into here. Then if the size already exceeds that param, just ignore and not push it to the slice.